src/rsz/src/RepairDesign.cc

```diff
@@ -55,14 +55,18 @@ RepairDesign::RepairDesign(Resizer* resizer) : resizer_(resizer)
 {
 }
 
-RepairDesign::~RepairDesign() = default;
+RepairDesign::~RepairDesign()
+{
+  delete pre_checks_;
+}
 
 void RepairDesign::init()
 {
   logger_ = resizer_->logger_;
   dbStaState::init(resizer_->sta_);
   db_network_ = resizer_->db_network_;
   dbu_ = resizer_->dbu_;
+  delete pre_checks_;
```

Comment:
Could we make this unique_ptr since assigning to a unique_ptr will auto free the currently store object. That way we can avoid specifying an explicit destructor for this class.